src/HtmlParser/Util.elm

```diff
@@ -390,3 +392,279 @@ toVirtualDomEach node =
 toAttribute : (String, String) -> Attribute msg
 toAttribute (name, value) =
   attribute name value
+
+{-| Converts nodes to virtual dom SVG nodes
+-}
+toVirtualDomSvg : List HtmlParser.Node -> List (Svg msg)
+toVirtualDomSvg nodes =
+  List.map toVirtualDomSvgEach nodes
+
+toVirtualDomSvgEach : HtmlParser.Node -> Svg msg
+toVirtualDomSvgEach node =
+  case node of
+    HtmlParser.Element name attrs children ->
+      Svg.node name (List.filterMap toSvgAttribute attrs) (toVirtualDomSvg children)
+
+    HtmlParser.Text s ->
+      text s
+
+    HtmlParser.Comment _ ->
+      text ""
+
+toSvgAttribute : (String, String) -> Maybe (Attribute msg)
+toSvgAttribute (name, value) =
+  case name of
```

Comment:
@coreyhaines I moved things around a few things to try to make it more attractive, any suggestions? My original solution was finding a Maybe (String -> Svg msg) using the case statement and then applying whatever I found to the value, but that would require some applicative-functor type stuff that I'm not sure how to do in Elm, and it wouldn't get rid of the millions of Just statements.